packages/worker/src/index.ts

```diff
@@ -5,60 +5,124 @@ import {
-		config.aws.localstackEndpoint,
-	);
+		const numberOfThreads = config.app.stage === 'PROD' ? 16 : 2;
 
-	if (isFailure(message)) {
-		return;
-	}
-	console.log(config.app.taskQueueUrl);
-
-	await publishTranscriptionOutput(
-		snsClient,
-		config.app.destinationTopicArns.transcriptionService,
-		{
-			id: 'test-id',
-			transcriptionSrt: 'test-srt',
-			languageCode: 'en',
-			userEmail: 'test@test.com',
-		},
-	);
+		const client = getClient(config.aws.region, config.aws.localstackEndpoint);
+		const message = await getNextMessage(client, config.app.taskQueueUrl);
+		const snsClient = getSNSClient(
+			config.aws.region,
+			config.aws.localstackEndpoint,
+		);
+
+		if (isFailure(message)) {
+			console.error(`Failed to fetch message due to ${message.errorMsg}`);
+			return;
+		}
+
+		if (!message.message) {
+			console.log('No messages available');
+			return;
+		}
 
-	if (message.message?.ReceiptHandle) {
 		const job = parseTranscriptJobMessage(message.message);
+
 		console.log(
 			`Fetched transcription job with id ${message.message.MessageId}}`,
 			job,
 		);
-		// wait for 3 minutes then delete the message to simulate the transcription
-		setTimeout(() => {
+
+		if (!job) {
+			console.error('Failed to parse job message', message);
+			return;
+		}
+
+		const fileToTranscribe = await getFileFromS3(config, job?.s3Key);
+
+		// docker container to run ffmpeg and whisper on file
+		const containerId = await createContainer(path.parse(fileToTranscribe).dir);
+
+		const ffmpegResult = await convertToWav(containerId, fileToTranscribe);
+
+		if (
+			message.message?.ReceiptHandle &&
+			ffmpegResult.duration &&
+			ffmpegResult.duration !== 0
+		) {
+			// Adding 300 seconds (5 minutes) and the file duration
+			// to allow time to load the whisper model
+			await changeMessageVisibility(
+				client,
+				config.app.taskQueueUrl,
+				message.message.ReceiptHandle,
+				ffmpegResult.duration + 300,
+			);
+		}
+
+		const text = await getTranscriptionText(
+			containerId,
+			ffmpegResult.wavPath,
+			fileToTranscribe,
+			numberOfThreads,
+		);
+
+		const transcriptionOutput: TranscriptionOutput = {
+			id: job.id,
+			transcriptionSrt: text,
+			languageCode: 'en',
+			userEmail: job.userEmail,
+		};
+
+		await publishTranscriptionOutput(
+			snsClient,
+			config.app.destinationTopicArns.transcriptionService,
+			transcriptionOutput,
+		);
+
+		if (message.message?.ReceiptHandle) {
 			console.log(`Deleting message ${message.message?.MessageId}`);
-			deleteMessage(
+			await deleteMessage(
 				client,
 				config.app.taskQueueUrl,
-				message.message?.ReceiptHandle as string,
+				message.message.ReceiptHandle,
 			);
-		}, dummyDelay * 1000);
+		}
+	} catch (error) {
+		const msg = 'Worker failed to complete';
+		console.error(msg, error);
 	}
 };
 
+const getFileFromS3 = async (config: TranscriptionConfig, s3Key: string) => {
+	const s3Client = getS3Client(config.aws.region);
+
+	const fileToTranscribe = await getFile(
```

Comment:
tiny thing but at the moment this is a fairly generic function so you could just call it `file` rather than `file ToTranscribe`